os/net/netmgr/netdev_mgr_internal.c

```diff
@@ -232,10 +233,12 @@ int nm_ifdown(struct netdev *dev)
 		return -1;
 	}
 
-	ret = dev->t_ops.wl->deinit(dev);
-	if (ret < 0) {
-		ndbg("fail to down driver interface\n");
-		return -2;
+	if (dev->type == NM_WIFI) {
+		ret = dev->t_ops.wl->deinit(dev);
+		if (ret < 0) {
+			ndbg("fail to down driver interface\n");
+			return -2;
+		}
 	}
```

Comment:
why don't you add logic about NM_ETHERNET deinit because nm_ifup() handle NM_ETHERNET initialization.